### PR TITLE
Enabled structured output

### PR DESCRIPTION
**Pull request by tianmu-li**
1. Combine logits from decode and prompt
2. Move logits to cpu
3. Apply bitmask on cpu logits
4. Move logits back to hpu
All of these are only triggered when doing structured output

### CODE REVIEW

**Comment by tianmu-li** [comment]
Accuracy from https://github.com/vllm-project/vllm/tree/main/benchmarks#%EF%B8%8F-example---structured-output-benchmark aligned with CUDA.

**Comment by tianmu-li** [comment]
@xuechendi Fixed issue with merged prefill and rebased

**Review by xuechendi — APPROVED** [review]

**Comment by xuechendi** [comment]
jenkins is failing due to known issue, merge this pr
